### PR TITLE
Add the `include-all-prereleases` input

### LINKED ISSUE #126: Feature request: Allow `dev` value for `version` which installs the latest alpha, beta or release candidate release
It would be great of aside from the current `nightly` value for the `version` parameter (which installs the very latest nightly version), a general `dev` value could be added which installs the latest alpha, beta or release candidate version. So basically the very latest tagged release, no matter the pre-release status.

### PR DESCRIPTION
**Pull request by SaschaMann**
fixes #126

### CODE REVIEW

File: __tests__/installer.test.ts [hunk 1]
```diff
@@ -66,6 +66,14 @@ describe('version matching tests', () => {
...
     })
 
+    describe('include-prereleases', () => {
+        it('Chooses the highest available version that matches the input including prereleases', () => {
+            expect(installer.getJuliaVersion(testVersions, '^1.2.0-0', true)).toEqual('1.3.0-rc4')
+            expect(installer.getJuliaVersion(testVersions, '1', true)).toEqual('1.3.0-rc4')
+            expect(installer.getJuliaVersion(testVersions, '^1.2.0-0', false)).toEqual('1.2.0')
+        })
+    })
+
```
Comment (SaschaMann) [inline __tests__/installer.test.ts:76]: I would recommend starting here for the review.

**Comment by SaschaMann** [comment]
cc @EwoutH

**Comment by DilumAluthge** [comment]
Can you explain why this is necessary? What's the use case here?

**Comment by DilumAluthge** [comment]
More specifically, when would you want to use this option but not want to just use either `^1.8.0-` or `~1.8.0-`?

**Comment by SaschaMann** [comment]
`^1.8.0-0` would skip over `1.9.0-rc1` and jump to `1.9.0`. With this input enabled, you could have a build that always uses the latest tagged release no matter what.

Are there many use cases for it? Not entirely convinced. It can save some time manually upgrading workflows I guess and it's a fairly straightforward change that doesn't introduce much complexity that's been requested by someone.

**Comment by DilumAluthge** [comment]
@ewoutH Could you elaborate on what your specific use case is here? What is the situation in which you want 1.8.0-rc1 or 1.9.0-rc1 but you don't want 1.9.0 if it's available?

**Comment by SaschaMann** [comment]
> What is the situation in which you want 1.8.0-rc1 or 1.9.0-rc1 but you don't want 1.9.0 if it's available?

I think I didn't describe it well: without this and a version 1.8.0-0, it would match the following versions once they are released: 1.8.0-rc1, 1.8.0, 1.9.0. With this flag, it would be 1.8.0-rc1, 1.8.0, 1.9.0-rc1, 1.9.0.

**Comment by DilumAluthge** [comment]
Ohhhh now I get it. I actually thought that `^1.8.0-0` would already have included 1.8.0-rc1, 1.8.0, 1.9.0-rc1, and 1.9.0. But since it doesn't, I think this is a good feature to add.

**Comment by DilumAluthge** [comment]
I don't know if `include-prereleases` is the best name for this feature? Because if the user passes `^1.8.0-0` and sets `include-prereleases: false`, it will still match `1.8.0-rc1` (which is a prerelease), right?

**Comment by DilumAluthge** [comment]
> I think I didn't describe it well: without this and a version 1.8.0-0, it would match the following versions once they are released: 1.8.0-rc1, 1.8.0, 1.9.0. With this flag, it would be 1.8.0-rc1, 1.8.0, 1.9.0-rc1, 1.9.0.

I think it would be good to include this exact example in the docs. This example is what made it "click" for me.

**Comment by SaschaMann** [comment]
> I don't know if include-prereleases is the best name for this feature? Because if the user passes ^1.8.0-0 and sets include-prereleases: false, it will still match 1.8.0-rc1 (which is a prerelease), right?

Yes, that's correct. Despite that it's also what node-semver uses. Do you have a suggestion?

> I actually thought that ^1.8.0-0 would already have included 1.8.0-rc1, 1.8.0, 1.9.0-rc1, and 1.9.0.

Same.

**Comment by DilumAluthge** [comment]
Hmmm. Maybe `always-include-prereleases` or `always-allow-prereleases`?

**Comment by EwoutH** [comment]
> @EwoutH Could you elaborate on what your specific use case is here?

My main use case is to have an early warning system in CI runs. So one job that always uses the latest Julia pre-release version, but is allowed to fail. So it can catch deprecations and warnings in Julia alpha, beta and rc versions.

**Comment by SaschaMann** [comment]
> Hmmm. Maybe `always-include-prereleases` or `always-allow-prereleases`?

Hm, those made me think the opposite of that is "sometimes" allowing prereleases, which isn't quite accurate. `-0` also "always" includes pre-releases, just not all of them. But maybe `include-all-prereleases`?

**Comment by DilumAluthge** [comment]
Yeah `include-all-prereleases` is definitely better than `include-prereleases` (in my opinion).

**Review by DilumAluthge — COMMENTED** [review]
I think this is good to go after renaming the input to `include-all-prereleases`?

**Comment by EwoutH** [comment]
I have one other suggestion, maybe we can create levels of pre-releases. For example:

- `include-prereleases: "all"` (or `"alpha"`) includes all pre-releases (alpha, beta, rc)
- `include-prereleases: "beta"` includes beta and release-candidate pre-releases
- `include-prereleases: "rc"` includes release-candidate pre-releases

**Comment by SaschaMann** [comment]
That would require us to handle the version resolution ourselves rather than passing it to node's semver library which would introduce too much complexity to the action for (imo) little gain.

**Review by DilumAluthge — APPROVED** [review]

**Comment by EwoutH** [comment]
Awesome, thanks!

**Comment by SaschaMann** [comment]
Tagged a release, should be available now :)

**Pull request merged**